nemo_curator/stages/text/embedders/base.py

```diff
@@ -62,33 +60,23 @@ def outputs(self) -> tuple[list[str], list[str]]:
     def setup(self, _: WorkerMetadata | None = None) -> None:
         """Load the model for inference."""
         self.model = AutoModel.from_pretrained(self.model_identifier, local_files_only=True)
-        self.model.eval()
-        self.model.to("cuda")
+        self.model.eval().to("cuda")
 
     def process_model_output(
         self, outputs: torch.Tensor, model_input_batch: dict[str, torch.Tensor] | None = None
     ) -> torch.Tensor:
         """Process model outputs to create embeddings."""
         if self.pooling == "mean_pooling":
-            return self._mean_pooling(outputs, model_input_batch[ATTENTION_MASK_COLUMN])
+            return self._mean_pooling(outputs, model_input_batch[ATTENTION_MASK_COLUMN]).cpu()
```

Comment:
Do we need to upcast/force float32 here too ?

Reply:
Same as above https://github.com/NVIDIA-NeMo/Curator/pull/1064#discussion_r2349553904